app/assets/javascripts/Components/dashboard.jsx

```diff
@@ -30,16 +34,66 @@ class Dashboard extends React.Component {
             backgroundColor: 'rgb(75, 192, 192)',
           },
         ],
-      }
+      },
+      gradeEntryFormChart2: {}
     };
   }
 
+  GradeEntryFormColumnBreakdownUpdateData = (data) => {
+    // Helper function to update the chart state
+    let label_list = Array();
+    for(let i = 0; i < 105; i+=5) {
+      label_list.push(i)
+    }
+    let datasets = Array()
+    for(let j = 0; j < data[0].length; j++){
+      datasets.push({
+        label: data[0][j],
+        data: data[1][j],
+        backgroundColor: colours[j],
+      })
+    }
+
+    this.setState({
+      gradeEntryFormChart2: {
+        data: {
+          labels: label_list,
+          datasets: datasets,
+        },
+        options: {
+          plugins: {
+            tooltip: {
+              callbacks: {
+                title: function (tooltipItems) {
+                  var baseNum = parseInt(tooltipItems[0].label);
+                  if (baseNum === 0) {
+                    return '0-5';
+                  } else {
+                    return (baseNum + 1) + '-' + (baseNum + 5);
+                  }
+                }
+              }
+            },
+          }
+        }
+      }
+    })
+  }
+
   componentDidUpdate(prevProps, prevState) {
     if (prevState.assessment_id !== this.state.assessment_id) {
       if (this.state.display_course_summary) {
         // TODO
       } else if (this.state.assessment_type === 'GradeEntryForm') {
-        // TODO
+        // Make an AJAX request to retrieve chart data
```

Comment:
In terms of structuring the code, you should really move this AJAX call into the helper method. That way, the helper method can be something like `getGradeEntryFormColumnBreakdown` and handle both the AJAX request and updating the component state.